### DISCUSSION TITLE
Conditional styling of rows

### DISCUSSION THREAD

**Discussion by DrMalchev** [UI5/webcomponents-react#5986] [General]
Hi,
is there a way to conditionally style rows in Analytical table. For example if we have a table with column Status, I want all rows with a given status to be grayed out.
Thanks

**Comment by Lukas742** [comment]
Hi @DrMalchev 

this is possible by creating a [custom plugin hook](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable-recipes--docs#how-to-add-your-own-plugin-hooks). Please see this [example](https://stackblitz.com/edit/ui5wcr-at-w8edph?file=src%2FApp.tsx,src%2Fdata.json,src%2Findex.css) as reference.

Also please keep in mind that colors are not accessible for all users, e.g. when navigating via screen reader only, users still need to know that they're currently on a row with semantic meaning, so you need to set the respective aria attributes accordingly.

```jsx
function useRowBgColor(hooks) {
  hooks.getRowProps.push((props, instance) => {
    const { row } = instance;
    // change color via inline style and value from the dataset
    if (row.original.background) {
      return [
        props,
        {
          style: {
            ...props.style,
            background: row.original.background,
          },
        },
      ];
    }
    if (row.index % 2) {
      // change color via inline style
      return [props, { style: { ...props.style, background: 'lightgreen' } }];
    }
    // change color via a CSS class
    return [props, { className: clsx(props.className, 'lightred') }];
  });
}
```
